### ISSUE TITLE
Matchpyramid not working for some text lengths

### ISSUE THREAD

**Issue by jibrilfrej** [NTMC-Community/MatchZoo#767]
### Describe the bug
MatchPyramid does not work for some text lengths: if I set the preprocessor fixed_length_left and fixed_length_right parameters to some values: the model crashes. 

I have mz version 2.1.0

I did not have this issue with the other models I tried ("ArcII", "KNRM", "DUET", "DRMMTKS")

### To Reproduce
```python
import matchzoo as mz

fixed_length_left=5
fixed_length_right=25
train_pack = mz.datasets.wiki_qa.load_data('train', task='ranking')
preprocessor = mz.preprocessors.BasicPreprocessor(fixed_length_left=fixed_length_left,
                                                  fixed_length_right=fixed_length_right)
model, preprocessor, data_generator_builder, embedding_matrix = mz.auto.prepare(
                    task=mz.tasks.Ranking(),
                    model_class=mz.models.match_pyramid.MatchPyramid,
                    data_pack=train_pack,
                    preprocessor=preprocessor
                    )
train_preprocessed = preprocessor.transform(train_pack,verbose=0)
train_gen = data_generator_builder.build(train_preprocessed)
model.fit_generator(train_gen, epochs=1)
```

### Describe your attempts
- I tried other values for fixed_length_left and fixed_length_right.
Here are the ones that failed : fixed_length_left=5,fixed_length_right=50
                                                fixed_length_left=5,fixed_length_right=100
                                                fixed_length_left=10,fixed_length_right=25
                                                fixed_length_left=6,fixed_length_right=24
                                                fixed_length_left=12,fixed_length_right=24

**Comment by uduse** [comment]
Thanks for the snippet, it correctly reproduces the bug. I gave a quick look but it seems the bug's not that obvious and I need more time for investigation.

**Comment by uduse** [comment]
Fixed in #768

**Comment by jibrilfrej** [comment]
Thanks !!
